manager/state/raft/raft.go

```diff
@@ -435,9 +497,36 @@ func (n *Node) Run(ctx context.Context) error {
 			// saveToStorage.
 			if !raft.IsEmptySnap(rd.Snapshot) {
 				// Load the snapshot data into the store
-				if err := n.restoreFromSnapshot(rd.Snapshot.Data, false); err != nil {
+				snapCluster, err := n.clusterSnapshot(rd.Snapshot.Data)
+				if err != nil {
 					log.G(ctx).WithError(err).Error("failed to restore from snapshot")
```

Comment:
Should this clear `snapCluster`? It seems bad to use it below if an error was returned.